Mathlib/NumberTheory/NumberField/House.lean

```diff
@@ -0,0 +1,349 @@
+/-
+Copyright (c) 2024 Michail Karatarakis. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Michail Karatarakis
+-/
+import Mathlib.NumberTheory.SiegelsLemma
+import Mathlib.NumberTheory.NumberField.CanonicalEmbedding.Basic
+
+/-!
+# House of an algebraic number
+This file defines the house of an algebraic number `α`, which is
+the largest modulus of its conjugates.
+
+## References
+* [D. Marcus, *Number Fields*][marcus1977number]
+* [Keng, H. L, *Introduction to number theory*][keng1982house]
+
+## Tagshouse
+number field, algebraic number, house
+-/
+
+variable {K : Type*} [Field K] [NumberField K]
+
+namespace NumberField
+
+noncomputable section
+
+open Module.Free FiniteDimensional NumberField.canonicalEmbedding Matrix Finset
+
+attribute [local instance] Matrix.seminormedAddCommGroup
+
+/-- The house of an algebraic number as the norm of its image by the canonical embedding.-/
+def house (α : K) : ℝ := ‖canonicalEmbedding K α‖
+
+theorem house_eq_sup' (α : K) :
+    house α = univ.sup' univ_nonempty (fun φ : K →+* ℂ ↦ ‖φ α‖₊) := by
+  rw [house, ← coe_nnnorm, nnnorm_eq, ← sup'_eq_sup univ_nonempty]
+
+theorem house_sum_le_sum_house {ι : Type _} (s : Finset ι) (α : ι → K) :
+    house (∑ i in s, α i) ≤ ∑ i in s, house (α i) := by
+  simp only [house, map_sum]; apply norm_sum_le_of_le; intros; rfl
+
+theorem house_nonneg (α : K) : 0 ≤ house α := norm_nonneg _
+
+theorem house_mul_le (α β : K) : house (α * β) ≤ house α * house β := by
+  simp only [house, _root_.map_mul]; apply norm_mul_le
+
+@[simp] theorem house_intCast (x : ℤ) : house (x : K) = |x| := by
+  simp only [house, map_intCast, Pi.intCast_def, pi_norm_const, Complex.norm_eq_abs,
+    Complex.abs_intCast, Int.cast_abs]
+
+variable (K)
+
+/-- An equivalence between the set of embeddings of `K` into `ℂ` and the
+  index set of the chosen basis of the ring of integers of `K`. -/
+abbrev equivReindex : (K →+* ℂ) ≃ (ChooseBasisIndex ℤ (𝓞 K)) := Fintype.equivOfCardEq <|
+  by rw [Embeddings.card, ← finrank_eq_card_chooseBasisIndex, RingOfIntegers.rank]
+
+/-- The basis matrix for the embeddings of `K` into `ℂ`. This matrix is formed by
+  taking the lattice basis vectors of `K` and reindexing them according to the
+  equivalence `equivReindex`, then transposing the resulting matrix. -/
+abbrev basisMatrix : Matrix (K →+* ℂ) (K →+* ℂ) ℂ :=
+  (of fun i ↦ latticeBasis K (equivReindex K i)).transpose
+
+lemma inv_mulVec_eq_vec {n : Type _} { α : Type _} [Fintype n] [DecidableEq n] [CommRing α]
+    {A : Matrix n n α} [Invertible A] {u v : n → α} (hM : u = A.mulVec v) : A⁻¹.mulVec u = v := by
+  rw [hM, mulVec_mulVec, inv_mul_of_invertible, one_mulVec]
+
+variable [DecidableEq (K →+* ℂ)]
+
+theorem det_of_basisMatrix_non_zero : (basisMatrix K).transpose.det ≠ 0 := by
```

Comment:
Similarly here, this has nothing to with `house`, right?

Also, I think a more natural statement is without the `transpose`.